tortoise/tortoise.go

```diff
@@ -1156,3 +1147,121 @@ func computeExpectedVoteWeight(atxdb atxDataProvider, epochWeight map[types.Epoc
 	}
 	return total, nil
 }
+
+// computeBallotWeight compute and assign ballot weight to the weights map.
+func computeBallotWeight(
+	atxdb atxDataProvider,
+	weights map[types.BallotID]*big.Float,
+	ballot *types.Ballot,
+	layerSize,
+	layersPerEpoch uint32,
+) (*big.Float, error) {
+	if ballot.EpochData != nil {
+		var (
+			total, weight uint64
+			foundWeight   bool
+		)
+		for _, atxid := range ballot.EpochData.ActiveSet {
```

Comment:
what's the rationale that we use the active set from the ref ballot instead of from our local DB?

can you provide links to the slack discussion for more context?

Reply:
i don't understand the question. we get weight from active set and store it in memory. what do you mean by local db?

Reply:
i meant `atxdb.GetEpochWeight()`
the slack thread you pointed out didn't help me tho. why should we assign weight according to their view and not the local view?

say our local atxdb has ATX of  weight 1000 for epoch 10. in the ballot, there is active set that weigh 2000 for epoch 10. 
we calculate the global threshold based on our local atxdb. in that case, we will assign too much weight for this ballot?

Reply:
GetEpochWeight returns weight of all atxs that are recorded for epoch. so if we have one block that has active set of 2000, and this block is valid, GetEpochWeight will always return at least 2000.

Specifically how it works during partition:
1. there are two separate activesets, each 1000
2. blocks in each partition specify activeset at the start of epoch and because we compute weight based on activeset, weight will remain stable even after partition is removed.
3. let's say partition lasted for an epoch, expected weight will be 1000 before rejoin and each partition will make progress independently
4. after rejoin expected weight will grow to 2000 (GetEpochWeight), but because we are using activeset neither of partitions will be able to make progress without each other weight in votes

Reply:
ok. thanks for the detailed explanation.